pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/EventService.java

```diff
@@ -0,0 +1,203 @@
+     * */
+    public static void sendParticipantEventToDss(String eventName, DDPInstance ddpInstance, @NotNull String ddpParticipantId) {
+        int ddpInstanceId = ddpInstance.getDdpInstanceIdAsInt();
+        if (!skippedParticipantEventDao.isParticipantEventSkipped(ddpParticipantId, eventName, ddpInstanceId)) {
+            boolean dssSuccessfullyTriggered = triggerDssWithEvent(eventName, ddpInstance, System.currentTimeMillis() / 1000,
+                    ddpParticipantId, ddpParticipantId, KitReasonType.NORMAL);
+            addParticipantEvent(eventName, ddpInstanceId, ddpParticipantId, dssSuccessfullyTriggered);
+        } else {
+            logger.info("Participant event was skipped for event %s for participant %s ".formatted(eventName, ddpParticipantId)
+                            .concat("based on participant_event table. Dss will not get triggered"));
+            //add this event to the EVENT_QUEUE table, but without triggering dss and flag EVENT_TRIGGERED = false
+            addParticipantEvent(eventName, ddpInstanceId, ddpParticipantId, false);
+        }
+    }
+
+    /**
+     * Sends a POST request to DSS to trigger the event for a kit, using the information in the KitDDPNotification object.
+     * */
+    @VisibleForTesting
+    protected static boolean triggerDssByKitEvent(KitDDPNotification kitDDPNotification) {
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceById(kitDDPNotification.getDdpInstanceId());
+        return triggerDssWithEvent(kitDDPNotification.getEventName(), ddpInstance,
+                kitDDPNotification.getDate() / 1000, kitDDPNotification.getParticipantId(),
+                kitDDPNotification.getDdpKitRequestId(), kitDDPNotification.getUploadReason());
+    }
+
+    /**
+     * Tries to send a POST request to DSS to trigger the event for a kit or a participant,
+     * It will try MAX_TRIES times before giving up and adding the event to the EVENT_QUEUE table.
+     * @param eventType - the type of event to be triggered, e.g. "SALIVA_RECEIVED"
+     * @param eventInfo - the information to be sent in the event, for kit events it is the "ddp_kit_request_id",
+     *      and for participant events it is the ddp_participant_id
+     * @return true if the request was successful, false otherwise
+     * */
+    @VisibleForTesting
+    protected static boolean triggerDssWithEvent(@NonNull String eventType, DDPInstance ddpInstance, long eventDate,
+                                        @NotNull String ddpParticipantId, @NotNull String eventInfo, KitReasonType reason) {
+        final long initialInterval = 100; // base delay in milliseconds
+        final double multiplier = 2.0; // exponential backoff multiplier
+
+        IntervalFunction intervalFn = IntervalFunction.ofExponentialBackoff(initialInterval, multiplier);
+        RetryConfig retryConfig = RetryConfig.custom()
+                .maxAttempts(MAX_TRIES)
+                .intervalFunction(intervalFn)
+                .build();
```

Comment:
I think the initial interval needs to be longer.  What are the typical response times when hitting this DSS endpoint?  If the response time is shorter than the typical time it takes for the dss route to complete, does this run the risk of making repeated attempts that will actually get through, then causing further issues?